internal/checkers/http_const.go

```diff
@@ -0,0 +1,128 @@
+package checkers
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+	"strings"
+
+	"golang.org/x/tools/go/analysis"
+)
+
+// HTTPConst detects situations like
+//
+//	assert.HTTPStatusCode(t, handler, "GET", "/index", nil, 200)
+//	assert.HTTPBodyContains(t, handler, "GET", "/index", nil, "counter")
+//
+// and requires
+//
+//	assert.HTTPStatusCode(t, handler, http.MethodGet, "/index", nil, http.StatusOK)
+//	assert.HTTPBodyContains(t, handler, http.MethodGet, "/index", nil, "counter")
+type HTTPConst struct{}
+
+// NewHTTPConst constructs Regexp checker.
+func NewHTTPConst() HTTPConst  { return HTTPConst{} }
+func (HTTPConst) Name() string { return "http-const" }
+
+func (checker HTTPConst) Check(_ *analysis.Pass, call *CallMeta) *analysis.Diagnostic {
+	var suggestedFixes []analysis.SuggestedFix
+	switch call.Fn.NameFTrimmed {
+	case "HTTPBody",
+		"HTTPBodyContains",
+		"HTTPBodyNotContains",
+		"HTTPError",
+		"HTTPRedirect",
+		"HTTPSuccess":
+
+		if len(call.Args) < 2 {
+			return nil
+		}
+		if suggestedFix := newHTTPMethodReplacement(call); suggestedFix != nil {
```

Comment:
```suggestion
suggestedFix := newHTTPMethodReplacement(call)
if suggestedFix == nil {
        return nil
}

suggestedFixes = append(suggestedFixes, *suggestedFix)

```

Reply:
Wait, are you sure suggestedFixes ia empty before returning?

Reply:
Maybe the fix for the extra and the potential bug is to remove all the else branch, including the return

Please check if there is no other return that could break the pending suggestedFixes